src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -39,30 +41,75 @@
 LOG_RUNNING_PROMPT = "This command usually takes minutes to run. Add '--verbose' parameter if needed."
 
 
-def spring_cloud_create(cmd, client, resource_group, name, location=None, sku=None, no_wait=False):
+def spring_cloud_create(cmd, client, resource_group, name, location=None, app_insights_key=None, app_insights=None,
+                        disable_distributed_tracing=None, sku=None, tags=None, no_wait=False):
     rg_location = _get_rg_location(cmd.cli_ctx, resource_group)
     if location is None:
         location = rg_location
 
     if sku is None:
         sku = "Standard"
-    full_sku = models.Sku(
-        name=_get_sku_name(sku),
-        tier=sku
-    )
-    resource = models.ServiceResource(location=location, sku=full_sku)
+    full_sku = models.Sku(name=_get_sku_name(sku), tier=sku)
+
+    properties = models.ClusterResourceProperties()
+    update_tracing_config(cmd, resource_group, name, location, properties,
+                          app_insights_key, app_insights, disable_distributed_tracing)
 
+    resource = models.ServiceResource(location=location, sku=full_sku, properties=properties, tags=tags)
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name=resource_group, service_name=name, resource=resource)
 
 
-def spring_cloud_update(cmd, client, resource_group, name, sku=None, no_wait=False):
-    full_sku = models.Sku(
-        name=_get_sku_name(sku),
-        tier=sku
-    )
-    resource = models.ServiceResource(sku=full_sku)
+def spring_cloud_update(cmd, client, resource_group, name, app_insights_key=None, app_insights=None,
+                        disable_distributed_tracing=None, sku=None, tags=None, no_wait=False):
+    updated_resource = models.ServiceResource()
+    update_app_insights = False
+    update_service_tags = False
+    update_service_sku = False
+
+    # update service sku
+    if sku is not None:
+        full_sku = models.Sku(name=_get_sku_name(sku), tier=sku)
+        updated_resource.sku = full_sku
+        update_service_sku = True
+
+    resource = client.get(resource_group, name)
+    location = resource.location
+    resource_properties = resource.properties
+    updated_resource_properties = models.ClusterResourceProperties()
+
+    app_insights_target_status = False
+    if app_insights is not None or app_insights_key is not None or disable_distributed_tracing is False:
+        app_insights_target_status = True
+        if resource_properties.trace.enabled is False:
+            update_app_insights = True
+    elif disable_distributed_tracing is True:
+        app_insights_target_status = False
+        if resource_properties.trace.enabled is True:
+            update_app_insights = True
```

Comment:
Based on the logic, if app_insights is not None, disable_distributed_tracing is True, it will be treated the same as disable_distributed_tracing is False.

I suggest to use `action='store_true'` for disable_distributed_tracing. When you want to enable distributed tracing, you just provide app_insights or app_insights_key. If you provide disable_distributed_tracing, it means true and can have higher priority or throw error when both disable_distributed_tracing and (app_insights or app_insights_key) are provided.

Reply:
Have fixed, thanks a lot